### PR TITLE
Switch OpenTracing for a OpenTelemetry test

### PR DESCRIPTION
**Pull request by mdujava**
Coexistence of OpenTelemetry and OpenTracing tests will work in case of reverting "[Remove depricated OpenTracing integration test](https://github.com/3scale-qe/3scale-tests/commit/cb51b26621e0eb0eb4f2324a2a189473d15203be)" commit.

### CODE REVIEW

File: testsuite/jaeger.py [hunk 1]
```diff
@@ -29,7 +29,35 @@ def traces(self, service: str, operation: str):
...
+[processors.batch]
+max_queue_size = 2048
+schedule_delay_millis = 5000
+max_export_batch_size = 512
+[service]
+name = "{service_name}" # Opentelemetry resource name
+""",
+        }
+
+    def apicast_config_open_tracing(self, configmap_name, service_name):
```
Comment (mganisin) [inline testsuite/jaeger.py]: Don't change eixsting  interfaces

File: testsuite/gateways/apicast/operator.py [hunk 2]
```diff
@@ -224,17 +224,43 @@ def connect_jaeger(self, jaeger):
...
+            apicast["openTelemetry"] = {
+                "enabled": True,
+                "tracingConfigSecretRef": {"name": secret_name},
+            }
+
+        self.apicast.modify_and_apply(_add_open_telemetry)
+        self.reload()
+        return self.name
+
+    def connect_open_tracing(self, jaeger):
```
Comment (mganisin) [inline testsuite/gateways/apicast/operator.py]: don't change existing interfaces

**Review by mganisin — CHANGES_REQUESTED** [review]

File: testsuite/gateways/apicast/operator.py [hunk 3]
```diff
@@ -215,7 +215,7 @@ def _update(apicast):
         self.apicast.modify_and_apply(_update)
         self.reload()
 
-    def connect_jaeger(self, jaeger):
+    def connect_open_telemetry(self, jaeger):
```
[the remark this answers is not in the document]
Comment (mdujava) [reply testsuite/gateways/apicast/operator.py:218]: Updated `apicast/system.py` and `apicast/template`.

File: testsuite/jaeger.py [hunk 4]
```diff
@@ -29,7 +29,35 @@ def traces(self, service: str, operation: str):
...
 
-    def apicast_config(self, configmap_name, service_name):
+    def apicast_config_open_telemetry(self, configmap_name, service_name):
+        """
+        :param configmap_name name of the configmap
+        :param service_name how apicast using this configmap will be named in jaeger
+        """
+        host, port = self.custom_config["reporter"]["localCollectorHostPort"].split(":")
+        return {
+            configmap_name: f"""# "otlp" is the only supported exporter in APIcast
```
[the remark this answers is not in the document]
Comment (mdujava) [reply testsuite/jaeger.py]: moved config to resource file

File: testsuite/tests/apicast/parameters/jaeger/test_open_telemetry_apicast_integration.py [hunk 5]
```diff
@@ -12,8 +12,16 @@
...
+def jaeger_service_name(staging_gateway, jaeger):
+    """
+    Deploys template apicast gateway configured with jaeger.
+    """
+    return staging_gateway.connect_open_telemetry(jaeger)
+
+
 @pytest.mark.issue("https://issues.redhat.com/browse/THREESCALE-5669")
-def test_jaeger_apicast_integration(api_client, jaeger, jaeger_service_name):
+def test_open_telemetry_apicast_integration(api_client, jaeger, jaeger_service_name):
```
Comment (mganisin) [inline testsuite/tests/apicast/parameters/jaeger/test_open_telemetry_apicast_integration.py:24]: no reference to relevant jira? (open telemetry support)

**Review by mganisin — APPROVED** [review]

**Review by mganisin — APPROVED** [review]

**Review by pehala — APPROVED** [review]

**Pull request merged**